app/services/downstream_service.rb

```diff
@@ -18,6 +18,11 @@ def self.update_draft_content_store(downstream_payload)
       message = "Can only send draft, published and unpublished items to draft content store"
       raise DownstreamInvalidStateError.new(message)
     end
+    if downstream_payload.state != "draft" && draft_at_base_path?(downstream_payload.base_path)
+      message = "Can't send #{downstream_payload.state} item to draft content store, as there is a draft occupying the same base path"
+      raise DownstreamInvalidStateError.new(message)
```

Comment:
Sure, it's not strictly a concurrency issue as it could happen any time you queued a published/unpublished item. 

I think it makes sense for an error to be raised here as it is an error situation, however it's in the concern of the workers on how to deal with it (currently log for indirect publishings and raise for direct ones).